.github/workflows/actions/database-backup/action.yml

```diff
@@ -0,0 +1,170 @@
+name: Backup AKS Database
+description: backs up AKS database to Azure Storage
+
+inputs:
+  environment:
+    description: "The name of the environment"
+    required: true
+  azure_credentials:
+    description: "JSON object containing a service principal that can read from Azure Key Vault"
+    required: true
+  restoreToProductionDataEnv:
+    required: true
+    type: boolean
+    default: false
+
+outputs:
+  backup_artifact:
+    description: "The backup artifact name"
+    value: ${{ inputs.environment }}-backup
+
+runs:
+  using: composite
+
+  steps:
+    - uses: actions/checkout@v4
+      id: Checkout
+
+    - name: Set KV environment variables
+      id: set_kv_env_vars
+      run: |
+        tf_vars_file=terraform/aks/workspace_variables/${{ inputs.environment }}_aks.tfvars.json
+        echo "INF_VAULT_NAME=$(jq -r '.inf_vault_name' ${tf_vars_file})" >> $GITHUB_ENV
+        echo "NAMESPACE=$(jq -r '.namespace' ${tf_vars_file})" >> $GITHUB_ENV
+        echo "APP_ENVIRONMENT=$(jq -r '.app_environment' ${tf_vars_file})" >> $GITHUB_ENV
+        echo "CLUSTER=$(jq -r '.cluster' ${tf_vars_file})" >> $GITHUB_ENV
+
+        if [ -z "$INF_VAULT_NAME" ]; then
+            echo "::error ::Failed to extract inf_vault_name from $TFVARS"
+            exit 1
+        fi
+
+        if [ -z "$NAMESPACE" ]; then
+            echo "::error ::Failed to extract namespace from $TFVARS"
+            exit 1
+        fi
+
+        if [ -z "$CLUSTER" ]; then
+            echo "::error ::Failed to extract cluster from $TFVARS"
+            exit 1
+        fi
+
+        if [ -z "$APP_ENVIRONMENT" ]; then
+            echo "::error ::Failed to extract app_environment from $TFVARS"
+            exit 1
+        fi
+
+        echo ::set-output name=cluster::$CLUSTER
+        echo ::set-output name=namespace::$NAMESPACE
+        echo ::set-output name=cluster::$CLUSTER
+        echo ::set-output name=app_environment::$APP_ENVIRONMENT
+      shell: bash
+      env:
+        TFVARS: ${{ inputs.tfvars }}
+      working-directory: terraform/aks
+
+    - uses: Azure/login@v1
+      with:
+        creds: ${{ inputs.azure_credentials }}
+
+    - uses: DfE-Digital/keyvault-yaml-secret@v1
+      id: keyvault-yaml-secret
```

Comment:
we are moving away from yaml secret. use key vault values
see example in https://github.com/DFE-Digital/schools-experience/blob/master/.github/workflows/build.yml#L77